lib/mix/tasks/install.ex

```diff
@@ -0,0 +1,194 @@
+defmodule Mix.Tasks.Beacon.Install do
+  @shortdoc "Generates beacon base files into the project"
+
+  @moduledoc """
+  Generates beacon necessary configurations.
+
+  Before running this command, make sure you commited all your changes to git,
+  beacuse it generates new files and modifies existing ones.
+
+    $ mix beacon.install --beacon-site "my-beacon-site"
+
+  The argument `beacon-site` defines the name of your beacon site and is
+  used to generate the necessary configuration files, defaults to `my-site`.
+
+  ## Arguments
+
+    * `--beacon-site` - The name of your beacon site, defaults to `my-site`
+  """
+  use Mix.Task
+
+  @switches [
+    beacon_site: :string
+  ]
+
+  def run(argv) do
+    if Mix.Project.umbrella?() do
+      Mix.raise("mix beacon.install can only be run inside an application directory")
+    end
+
+    {options, _parsed} = OptionParser.parse!(argv, strict: @switches)
+
+    bindings = build_context_bindings(options)
+
+    # Add Beacon.Repo to config.exs
+    config_file = config_file("config.exs")
+    maybe_add_beacon_repo(config_file, File.read!(config_file))
+
+    # Add Beacon.Repo database config to dev.exs and prod.exs
+    dev_config_file = config_file("dev.exs")
+    prod_config_file = config_file("prod.exs")
+
+    maybe_add_beacon_repo_config(dev_config_file, bindings)
+    maybe_add_beacon_repo_config(prod_config_file, bindings)
+
+    # Create BeaconDataSource file and config
+    maybe_create_beacon_data_source_file(bindings)
+    maybe_add_beacon_data_source_to_config(config_file, File.read!(config_file), bindings)
```

Comment:
Maybe move the reading of the file to the function?

```suggestion
    maybe_add_beacon_data_source_to_config(config_file, bindings)
```